DATAFLOW.md

```diff
@@ -1,28 +1,27 @@
+   
+   The upload to the "input" bucket triggers a Document Processing Workflow managed by Cloud Composer.  
 1. **Validate the documents:**
    Documents that fail validation are redirected to a separate "reject" bucket within Google Cloud Storage for further review and handling.
 1. **Initiate Processing Jobs:**
    The workflow initiates specific processing jobs based on the document type:
-   `PDF Classifier (Cloud Run -> Doc AI Custom Classifier):` Identify PDF documents that contain forms.
-   `Forms Processor (Cloud Run -> Doc AI Form Parser):` Processes form documents(PDF).
-   `Excel, Outlook Processor (Cloud Run):` Processes Excel and Outlook files.
-1. **Extract and store content:**
-   Each processor extracts relevant content from the documents. The extracted content (PDF, docx, txt) is stored in a “process” bucket within Google Cloud Storage.
-1. **Create and store document metadata for Excel and Outlook documents:**
-   The extracted document metadata is stored in a BigQuery table.
+   - `Document Classifier (Cloud Run -> Doc AI Custom Classifier):`  Identify PDF document type (Any form, bank statements, invoices, pay slip, lending doc and utility bills)  
+   - `Specialized Parser (Cloud Run -> Doc AI Form and Specialied Parser):` Processes forms and documents (Any form, bank statements, invoices, pay slip, lending doc and utility bills)  
+   - `Excel, Outlook Processor (Cloud Run):` Processes Excel and Outlook files.  
 1. **Utilize Document AI Batch API:**
-   The Document Processing Workflow leverages the Document AI Batch API for asynchronous processing of large document volumes.
-1. **Create and store document metadata for forms in PDF documents:**
-   The extracted document metadata is stored in a BigQuery table.
+   The Specialized Parser Job leverages Document AI Batch API for asynchronous processing of large document volumes.
+1. **Extract and store data elements:**
+   The Specialized Parser Job uses Document AI Batch API to extract data elements from the documents and stores the extracted data elements in AlloyDB, BigQuery and Firestore.  
+1. **Create and store document metadata in BigQuery for Search and Summarization:**
+   Document processsing jobs creates and stores document metadata in a BigQuery table. Document metadata is required by Agent Builder for indexing the documents. Document metadata contains the uri to the documents stored in the "process" bucket in Google Cloud Storage.
+1. **Create and store processed documents in Cloud Storage Bucket for Search and Summarization:**
+   The processed documents and document metadata files are also stored in the "process" bucket in Google Cloud Storage for search and summarization.
 1. **Data Serving:**
    The processed and stored content from the "process" bucket is imported and indexed in Agent Builder’s BigQuery Data Store.
```

Comment:
"GCS data is imported... to Agent Builder's **BigQuery** Data Store" --> is this correct?
I thought the data store could be either GCS-based or BQ-based depending on the input of source files.  The diagram shows both but the text makes it sound like GCS inputs are converted to a BQ-based datastore. 

To make clearer, how about something that generalizes across both sources like "Data and metadata from the processed documents is imported into an Agent Builder Data Store"

Reply:
I'll just say that documents are kept in GCS, but data extracted from the document is kept in BQ and AlloyDB, plus metadata in BQ, plus ingested in Agent builder, which is the secret sauce (vectorized version of the document).

So combination of all.